src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -122,68 +117,140 @@ public static void tearDown() {
         apiUser = Persistence.apiUsers.getById(apiUser.id);
         if (apiUser != null) apiUser.delete();
         otpUser = Persistence.otpUsers.getById(otpUser.id);
-        if (otpUser != null) otpUser.delete();
+        if (otpUser != null) otpUser.delete(false);
     }
 
     /**
      * Tests to confirm that an otp user, related monitored trip and plan can be created and deleted leaving no orphaned
      * records. This also includes Auth0 users if auth is enabled.
      */
     @Test
-    public void canSimulateApiUserFlow() throws IOException, URISyntaxException {
-        // create otp user as api user
-        HttpResponse<String> createUserResponse = mockAuthenticatedPost("api/secure/user",
+    public void canSimulateApiUserFlow() throws URISyntaxException {
+
+        // obtain Auth0 token for Api user.
+        String endpoint = String.format("api/secure/application/authenticate?username=%s&password=%s",
+            apiUser.email,
+            TEMP_AUTH0_USER_PASSWORD);
+        HttpResponse<String> getTokenResponse = mockAuthenticatedRequest(endpoint,
+            "",
             apiUser,
-            JsonUtils.toJson(otpUser)
+            HttpUtils.REQUEST_METHOD.POST,
+            false
         );
-        assertEquals(HttpStatus.OK_200, createUserResponse.statusCode());
-        OtpUser otpUserResponse = JsonUtils.getPOJOFromJSON(createUserResponse.body(), OtpUser.class);
+        LOG.info(getTokenResponse.body());
+        assertEquals(HttpStatus.OK_200, getTokenResponse.statusCode());
+        TokenHolder tokenHolder = JsonUtils.getPOJOFromJSON(getTokenResponse.body(), TokenHolder.class);
 
-        // Generate mock response to create monitored trip
-        // FIXME: Refactor once the method to store monitored trip is refactored.
-        String mockResponse = FileUtils.getFileContents(
-            "src/test/resources/org/opentripplanner/middleware/persistence/planResponse.json"
+        // Define the header values to be used in requests from this point forward.
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("Authorization", "Bearer " + tokenHolder.getAccessToken());
+        headers.put("x-api-key", apiUser.apiKeys.get(0).value);
+
+
+        // create an Otp user authenticating as an Api user.
+        HttpResponse<String> createUserResponse = authenticatedRequest("api/secure/user",
+            JsonUtils.toJson(otpUser),
+            headers,
+            HttpUtils.REQUEST_METHOD.POST
         );
-        OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse(mockResponse, DEFAULT_PLAN_URI);
+
+        assertEquals(HttpStatus.OK_200, createUserResponse.statusCode());
+
+        // Attempt to create a monitored trip for an Otp user authenticating as an Otp user. This will fail because the
```

Comment:
"for an Otp user authenticating as an Otp user."... what does that mean?

Reply:
Updated.